cpp/include/cudf/io/writers.hpp

```diff
@@ -94,6 +94,75 @@ class writer {
 
 }  // namespace orc
 
+
+//! Parquet format
+namespace parquet {
+
+/**
+ * @brief Statistics granularity type
+ */
+enum statistics_freq {
+  statistics_none = 0,     //!< No column statistics
+  statistics_rowgroup = 1, //!< Per-Rowgroup column statistics
+  statistics_page = 2,     //!< Per-page column statistics
+};
+
+/**
+ * @brief Options for the parquet writer.
+ */
+struct writer_options {
+  compression_type compression = compression_type::AUTO;
+  int stats_granularity = 1;
```

Comment:
What is `stats_granularity`?  Is that the same as the `enum statistics_freq`?

Reply:
Yes, they are both the same as statistics_freq (in fact, I initially had it as statistics_freq type), but because the enum is defined in a header that isn't shared (it is used in functions.hpp, which doesn't include writer.hpp), it didn't work if I used the enum unless I either move the definition of statistics_freq to types.hpp (like for compression_type), or include writers.hpp in functions.hpp. I was thinking of revisiting this when adding statistics to the ORC writer (maybe define format-agnostic statistics_freq values like none/minimal/maximum, or just use the parquet definitions for ORC where rowgroup=stripe.

Reply:
Moved the statistics_freq definition to types.hpp. Will probably just use the parquet definitions for ORC statistics (parquet rowgroup == ORC stripe).